contrib/coredns/policy/policytap/policytap.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto2";
+
+import "github.com/dnstap/golang-dnstap/dnstap.proto";
+
+package policytap;
+
+message DnstapAttribute {
+	required string id = 1;
+	required string type = 2;
+	required string value = 3;
+}
+
+message PolicyHitMessage {
+
+	// policy actions of interest
+	enum PolicyActionType {
+		POLICY_ACTION_DENY = 1;
+		POLICY_ACTION_PASSTHROUGH = 2;
+		POLICY_ACTION_NXDOMAIN = 3;
+		POLICY_ACTION_REDIRECT = 4;
+		POLICY_ACTION_REFUSE = 5;
+	}
+
+	// two types of policy trigger
+	enum PolicyTriggerType {
+		POLICY_TRIGGER_ADDRESS = 1;
+		POLICY_TRIGGER_DOMAIN = 2;
+	}
+
+	// Policy hit time, in seconds and nanoseconds since the UNIX epoch.
+	required uint64 time_sec = 1;
+	required fixed32 time_nsec = 2;
+
+	// DNS query/transaction id; this is a 16-bit value, but
+	// ProtocolBuffers interface defines only longer integers.
+	required uint32 query_id = 3;
+
+	// Policy Trigger Type -- domain name or address
+	required PolicyTriggerType trigger_type = 4;
+
```

Comment:
Where are the trigger_name and trigger_category?  I think PDP will be able to provide these 2 info.  We should define them here as well.

Reply:
trigger_name and trigger_category should be configured in PAP. If configured, PDP will be return these attributes as obligations, and they will be copied to PolicyHitMessage.attributes